### PR TITLE
Using federated auth for ACS tests

### PR DESCRIPTION
**Pull request by rajuanitha88**
Changes:
In order to move away from Secret based authentication, we want to adopt FIC.
For the same azure-sdk team has added the support and we need to use service connection for our live test pipelines to access our test resources.

### CODE REVIEW

File: sdk/communication/azure-communication-identity/tests.yml [hunk 1]
```diff
@@ -8,21 +8,16 @@ extends:
...
         - TestSamples=.*/true
+      UseFederatedAuth: true
       CloudConfig:
         Public:
+          ServiceConnection: azure-sdk-tests
+          SubscriptionConfigurationFilePaths:
+            - eng/common/TestResources/sub-config/AzurePublicMsft.json
           SubscriptionConfigurations:
-            - $(sub-config-azure-cloud-test-resources)
             - $(sub-config-communication-services-cloud-test-resources-common)
```
Comment (mjafferi-msft) [inline sdk/communication/azure-communication-identity/tests.yml:18]: Can we use the new secret we created just for CTE here as well and remove the other ones ?

**Review by mjafferi-msft — APPROVED** [review]
Just 1 small comment to check if we can use CTE secret here as well to maintain consistency across all the platforms, other looks good to me.

File: sdk/communication/azure-communication-identity/tests.yml (hunk 1 above)
Comment (rajuanitha88) [reply sdk/communication/azure-communication-identity/tests.yml:18]: Keeping it consistent across SDKs (and choosing the older keyvault instread of CTE)

**Review by maximrytych-ms — APPROVED** [review]

**Review by mikehang-msft — APPROVED** [review]

**Review by weshaggard — APPROVED** [review]

**Review by LuChen-Microsoft — APPROVED** [review]

**Pull request merged**